src/models/StakeRequestRepository.ts

```diff
@@ -16,17 +16,34 @@
 
 /* eslint-disable class-methods-use-this */
 
-import {
-  DataTypes, Model, InitOptions,
-} from 'sequelize';
+import { DataTypes, Model, InitOptions } from 'sequelize';
 import BigNumber from 'bignumber.js';
+import { MessageModel } from './MessageRepository';
+import Subject from '../observer/Subject';
 
-import {
-  MessageModel,
-  Message,
-} from './MessageRepository';
+class StakeRequestModel extends Model {
```

Comment:
It becomes difficult to understand `StakeRequestModel`, `StakeRequestAttrubutes` and `StakeRequest`.
Should we make`StakeRequest` as a class? This repository can work on `StakeRequest` object instead of interfaces.
So it is always StakeRequest object that is used in the application.

What do you think?

Reply:
StakeRequestModel is not exported. StakeRequestAttributes and StakeRequest are exposed. StakeRequestAttributes acts as an interface for a data that needs to be passed to create a StakeRequest object. StakeRequest object is used through the application. I am not sure what would give us making StakeRequest a class. What member will have that class?

Reply:
When I call `GatewayRepository.create` with object `a`, I get back object `b`.
Ideally, it should return me object `a` with updated properties.

Reply:
Why, `ideally`? An input and output are of different types.
In general, we should not modify the object passed as an argument, and instead return a new object.

Reply:
> Why, `ideally`? An input and output are of different types.
> In general, we should not modify the object passed as an argument, and instead return a new object.


The way I have seen it is it is just an input, and it saves the objects, and updates the properties like updated at/created at. As discussed this is the behavior of `save` and not `create`.

We can go with your approach.